src/session/session_dhandle.c

```diff
@@ -575,9 +575,19 @@ __wt_session_get_btree_ckpt(WT_SESSION_IMPL *session, const char *uri, const cha
              */
 
             if (first_snapshot_time != snapshot_time || ds_time > snapshot_time ||
-              hs_time > snapshot_time || stable_time > snapshot_time || oldest_time > snapshot_time)
-                ret = __wt_set_return(session, EBUSY);
-            else {
+              hs_time > snapshot_time || stable_time > snapshot_time ||
+              oldest_time > snapshot_time) {
```

Comment:
I think (oldest_time > snapshot_time) is not problematic. My opinion is that even if the oldest timestamp is moved beyond the snapshot_time of the checkpoint, we should be still able to open the checkpoint, I don't think we should return EBUSY.

Reply:
That's a good point @bvpvamsikrishna. I have opened [WT-10718](https://jira.mongodb.org/browse/WT-10718) to investigate this.